`mix test` failed after merging #42 get_node_info

`mix test` failed after merging #42 get_node_info.

---
Report on GitHub Actions:
https://github.com/rclex/rclex/runs/3043218395

---

The error is varied with ROSDISTRO.

message on dashing
```
Compiling 6 files (.ex)
Generated rclex app
finished rcl_init
finished rcl_init
.publishers finish
subscribers finish
node finish
node finish
.

Finished in 0.07 seconds
2 tests, 0 failures

Randomized with seed 960469
double free or corruption (fasttop)
/__w/_temp/14b5c79e-865a-4473-a19e-cabce5b3f5d2.sh: line 3:  2113 Aborted                 (core dumped) mix test
Error: Process completed with exit code 134.
```

message on foxy
```
Compiling 6 files (.ex)
Generated rclex app
finished rcl_init
.finished rcl_init


  1) test rcl_get_topic_names_and_types (RclexTest)
     test/rclex_test.exs:18
     ** (MatchError) no match of right hand side value: [{'/testtopic', ['std_msgs/msg/String']}]
     code: [ name_and_types_tuple_1 | [ name_and_types_tuple_2 ] ]= names_and_types_tuple_list
     stacktrace:
       test/rclex_test.exs:29: (test)



Finished in 0.1 seconds
2 tests, 1 failure

Randomized with seed 786112
```

--- a/lib/rclex.ex
+++ b/lib/rclex.ex
@@ -89,7 +89,7 @@ defmodule Rclex do
     raise "haha"
   end
 
-    @doc """
+  @doc """
       return char 
   """
 

--- a/test/rclex_test.exs
+++ b/test/rclex_test.exs
@@ -10,8 +10,9 @@ defmodule RclexTest do
     [node_1, node_2] = node_list
 
     assert Rclex.rcl_node_get_name(node_1) == 'test_node1', 'first node name is test_node1'
-
     assert Rclex.rcl_node_get_name(node_2) == 'test_node2', 'second node name is test_node2'
+
+    Rclex.node_finish(node_list)
   end
 
   # -----------------------graph_nif.c--------------------------

--- a/test/rclex_test.exs
+++ b/test/rclex_test.exs
@@ -1,51 +1,51 @@
 defmodule RclexTest do
-    use ExUnit.Case
+  use ExUnit.Case
 
-    # -----------------------node_nif.c--------------------------
-    test "rcl_node_get_name" do
-        context = Rclex.rclexinit
-        
-        node_name = 'test_node'
-        node_list = Rclex.create_nodes(context,node_name,2)
-        [node_1, node_2] = node_list
+  # -----------------------node_nif.c--------------------------
+  test "rcl_node_get_name" do
+    context = Rclex.rclexinit()
 
-        assert Rclex.rcl_node_get_name(node_1) == 'test_node1', 'first node name is test_node1'
+    node_name = 'test_node'
+    node_list = Rclex.create_nodes(context, node_name, 2)
+    [node_1, node_2] = node_list
 
-        assert Rclex.rcl_node_get_name(node_2) == 'test_node2', 'second node name is test_node2'
-    end
+    assert Rclex.rcl_node_get_name(node_1) == 'test_node1', 'first node name is test_node1'
 
-    # -----------------------graph_nif.c--------------------------
-    test "rcl_get_topic_names_and_types" do
-        context = Rclex.rclexinit
-        node_list = Rclex.create_nodes(context,'test_pub_node',1)
+    assert Rclex.rcl_node_get_name(node_2) == 'test_node2', 'second node name is test_node2'
+  end
 
-        publisher_list = Rclex.create_publishers(node_list,'testtopic',:single)
-        node_list_2 = Rclex.create_nodes(context,'test_sub_node',1)
-        subscriber_list = Rclex.create_subscribers(node_list_2, 'testtopic', :single)
-        node = hd node_list
+  # -----------------------graph_nif.c--------------------------
+  test "rcl_get_topic_names_and_types" do
+    context = Rclex.rclexinit()
+    node_list = Rclex.create_nodes(context, 'test_pub_node', 1)
 
-        names_and_types_tuple_list = Rclex.rcl_get_topic_names_and_types(node,Rclex.rcl_get_default_allocator,false)
+    publisher_list = Rclex.create_publishers(node_list, 'testtopic', :single)
+    node_list_2 = Rclex.create_nodes(context, 'test_sub_node', 1)
+    subscriber_list = Rclex.create_subscribers(node_list_2, 'testtopic', :single)
+    node = hd(node_list)
 
-        [ name_and_types_tuple_1 | [ name_and_types_tuple_2 ] ]= names_and_types_tuple_list
+    names_and_types_tuple_list =
+      Rclex.rcl_get_topic_names_and_types(node, Rclex.rcl_get_default_allocator(), false)
 
-        name_1 = elem(name_and_types_tuple_1,0)
-        types_list_1 = elem(name_and_types_tuple_1,1)
+    [name_and_types_tuple_1 | [name_and_types_tuple_2]] = names_and_types_tuple_list
 
-        assert name_1 == '/rosout', 'first topic name is rosout'
-        type_1 = hd types_list_1
-        assert type_1 == 'rcl_interfaces/msg/Log', 'first topic type is rcl_interfaces/msg/Log' 
-        
-        name_2 = elem(name_and_types_tuple_2,0)
-        types_list_2 = elem(name_and_types_tuple_2,1)
-        
-        assert name_2 == '/testtopic', 'second topic name is testtopic'
-        type_2 = hd types_list_2
-        assert  type_2 == 'std_msgs/msg/String', 'second topic type is std_msgs/msg/String' 
+    name_1 = elem(name_and_types_tuple_1, 0)
+    types_list_1 = elem(name_and_types_tuple_1, 1)
 
-        Rclex.publisher_finish(publisher_list,node_list)
-        Rclex.subscriber_finish(subscriber_list,node_list_2)
-        Rclex.node_finish(node_list)
-        Rclex.node_finish(node_list_2)
+    assert name_1 == '/rosout', 'first topic name is rosout'
+    type_1 = hd(types_list_1)
+    assert type_1 == 'rcl_interfaces/msg/Log', 'first topic type is rcl_interfaces/msg/Log'
 
-    end
+    name_2 = elem(name_and_types_tuple_2, 0)
+    types_list_2 = elem(name_and_types_tuple_2, 1)
+
+    assert name_2 == '/testtopic', 'second topic name is testtopic'
+    type_2 = hd(types_list_2)
+    assert type_2 == 'std_msgs/msg/String', 'second topic type is std_msgs/msg/String'
+
+    Rclex.publisher_finish(publisher_list, node_list)
+    Rclex.subscriber_finish(subscriber_list, node_list_2)
+    Rclex.node_finish(node_list)
+    Rclex.node_finish(node_list_2)
+  end
 end

--- a/test/rclex_test.exs
+++ b/test/rclex_test.exs
@@ -18,8 +18,8 @@ defmodule RclexTest do
   # -----------------------graph_nif.c--------------------------
   test "rcl_get_topic_names_and_types" do
     context = Rclex.rclexinit()
-    node_list = Rclex.create_nodes(context, 'test_pub_node', 1)
 
+    node_list = Rclex.create_nodes(context, 'test_pub_node', 1)
     publisher_list = Rclex.create_publishers(node_list, 'testtopic', :single)
     node_list_2 = Rclex.create_nodes(context, 'test_sub_node', 1)
     subscriber_list = Rclex.create_subscribers(node_list_2, 'testtopic', :single)
@@ -28,21 +28,14 @@ defmodule RclexTest do
     names_and_types_tuple_list =
       Rclex.rcl_get_topic_names_and_types(node, Rclex.rcl_get_default_allocator(), false)
 
-    [name_and_types_tuple_1 | [name_and_types_tuple_2]] = names_and_types_tuple_list
+    name_and_types_tuple = List.last(names_and_types_tuple_list)
 
-    name_1 = elem(name_and_types_tuple_1, 0)
-    types_list_1 = elem(name_and_types_tuple_1, 1)
+    name = elem(name_and_types_tuple, 0)
+    types_list = elem(name_and_types_tuple, 1)
 
-    assert name_1 == '/rosout', 'first topic name is rosout'
-    type_1 = hd(types_list_1)
-    assert type_1 == 'rcl_interfaces/msg/Log', 'first topic type is rcl_interfaces/msg/Log'
-
-    name_2 = elem(name_and_types_tuple_2, 0)
-    types_list_2 = elem(name_and_types_tuple_2, 1)
-
-    assert name_2 == '/testtopic', 'second topic name is testtopic'
-    type_2 = hd(types_list_2)
-    assert type_2 == 'std_msgs/msg/String', 'second topic type is std_msgs/msg/String'
+    assert name == '/testtopic', 'topic name is testtopic'
+    type = hd(types_list)
+    assert type == 'std_msgs/msg/String', 'topic type is std_msgs/msg/String'
 
     Rclex.publisher_finish(publisher_list, node_list)
     Rclex.subscriber_finish(subscriber_list, node_list_2)

